session.go

```diff
@@ -92,6 +95,7 @@ func newSession(config *Config, conn io.ReadWriteCloser, client bool) *Session {
 		streams:    make(map[uint32]*Stream),
 		synCh:      make(chan struct{}, config.AcceptBacklog),
 		acceptCh:   make(chan *Stream, config.AcceptBacklog),
+		goAwayCh: make(chan struct{}, 1),
```

Comment:
Little bit of a formatting fail (needs to be indented). Also, `make(chan struct{})` should be sufficient.